boot/setup.sh

```diff
@@ -28,14 +28,23 @@ mkdir -p /var/smartdc/$role
 
 /usr/bin/chown -R root:root /opt/smartdc
 
-echo "Generating SSL Certificate"
-mkdir -p /opt/smartdc/$role/ssl
-/opt/local/bin/openssl req -x509 -nodes -subj '/CN=*' -newkey rsa:2048 \
-    -keyout /opt/smartdc/$role/ssl/key.pem \
-    -out /opt/smartdc/$role/ssl/cert.pem -days 365
-
-cat /opt/smartdc/$role/ssl/cert.pem > /opt/smartdc/$role/ssl/stud.pem
-cat /opt/smartdc/$role/ssl/key.pem >> /opt/smartdc/$role/ssl/stud.pem
+# Mount the delegated dataset at /data
+zfs set mountpoint=/data zones/$(zonename)/data
+
+function setup_tls_certificate() {
+    if [[ -f /data/tls/key.pem && -f /data/tls/cert.pem ]]; then
+        echo "TLS Certificate Exists"
+    else
+        echo "Generating TLS Certificate"
+        mkdir -p /data/tls
+        /opt/local/bin/openssl req -x509 -nodes -subj '/CN=*' \
```

Comment:
This supposed to be a self-signed certificate? Add some comments that this is used as a fallback self-signed TLS certificate.

Also - maybe set expiry days to 3650? It's annoying when self-signed testing certs expire...

Reply:
Yeah, self signed. And good call on the expiry.

Reply:
It should be self-signed when the zone is created. It's expected it that operators will replace it either manually (if they buy one) or with triton-dehydrated.